python/tvm/autotvm/tuner/xgboost_cost_model.py

```diff
@@ -243,18 +243,27 @@ def fit_log(self, records, plan_size, min_seed_records=500):
         else:
             raise RuntimeError("Invalid feature type: " + self.fea_type)
         result = pool.map_with_error_catching(feature_extract_func, data)
+        result = list(result) # store results so we can iterate through them twice
 
-        # filter out feature with different shapes
-        fea_len = len(self._get_feature([0])[0])
+        # get maximum feature length
+        fea_len = -1
+        for res in result:
+            if res.status != StatusKind.COMPLETE:
+                continue
+            x, _ = res.value
+            fea_len = max(fea_len, x.shape[0])
 
         xs, ys = [], []
         for res in result:
             if res.status != StatusKind.COMPLETE:
                 continue
             x, y = res.value
-            if len(x) == fea_len:
```

Comment:
Hmm, I think previously glossed over the significance of this. Before, it would only use features from those with the same shape as the first entry to fit the model so basically loops with the same number of axis. So this is actually quite a major change is it not? 

The model is at the task level, so how did it handle configs with different size features between training rounds?